content/v2.0/application/settings.md

```diff
@@ -0,0 +1,266 @@
+
+```ruby
+# config/settings.rb
+
+# frozen_string_literal: true
+
+module Bookshelf
+  class Settings < Hanami::Settings
+    setting :analytics_enabled, constructor: Types::Params::Bool
+    setting :max_cart_items, constructor: Types::Params::Integer
+  end
+end
+```
+
+```shell
+bundle exec hanami server
+
+! Unable to load application: Hanami::Settings::InvalidSettingsError: Could not initialize settings. The following settings were invalid:
+
+analytics_enabled:  cannot be coerced to false
+max_cart_items: can’t convert nil into Integer
+```
+
+The same exception will be raised if a setting can't be correctly coerced:
+
+```shell
+ANALYTICS_ENABLED=true MAX_CART_ITEMS="not coerceable to integer"  bundle exec hanami server
+
+! Unable to load application: Hanami::Settings::InvalidSettingsError: Could not initialize settings. The following settings were invalid:
+
+max_cart_items: invalid value for Integer(): "not coerceable to integer"
+```
+
+To make a setting optional, use `optional` to allow `nil` values:
+
+```ruby
+# config/settings.rb
+
+# frozen_string_literal: true
+
+module Bookshelf
+  class Settings < Hanami::Settings
+    setting :analytics_enabled, constructor: Types::Params::Bool.optional
+    setting :max_cart_items, constructor: Types::Params::Integer.optional
+  end
+end
+```
+
+## Default values
+
+Settings can also specify defaults to be used in the absence of the relevant environment variable.
+
+```ruby
+# config/settings.rb
+
+# frozen_string_literal: true
+
+module Bookshelf
+  class Settings < Hanami::Settings
+    setting :redis_url, default: "redis://localhost:6379", constructor: Types::String
+    setting :analytics_enabled, default: false, constructor: Types::Params::Bool
+    setting :max_cart_items, default: 100, constructor: Types::Params::Integer
+  end
+end
+```
+
+## Constraints
+
+To enforce additional contraints on settings, you can use a [dry-types](https://dry-rb.org/gems/dry-types) constraint in your constructor.
+
+Here, the value of the `session_secret` must be at least 32 characters, while the value of the `from_email` setting must satisfy the EMAIL_FORMAT regular expression:
+
+```ruby
+# config/settings.rb
+
+# frozen_string_literal: true
+
+module Bookshelf
+  class Settings < Hanami::Settings
+    EMAIL_FORMAT = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z]+)*\.[a-z]+\z/i
+
+    setting :session_secret, constructor: Types::String.constrained(min_size: 32)
+    setting :from_email, constructor: Types::String.constrained(format: EMAIL_FORMAT)
+  end
+end
+```
+
+## Using settings within your application
+
+Hanami makes settings available within your application through a built-in settings provider.
+
+```ruby
+Hanami.app["settings"]
+```
+
+### Accessing settings within components
+
+To access settings within one of your components, use Hanami's Deps mixin:
```

Comment:
```suggestion
To access settings within one of your components, use the Deps mixin:
```

Do we need "Hanami's" here, since we're already in the Hanami guides? I actually don't know, keen for your thoughts.

Reply:
Yeah, we don't need Hanami in this case I think. Especially with the link to the docs on it below.